pkg/cmd/server/admin/create_servercert.go

```diff
@@ -76,7 +76,7 @@ func NewCommandCreateServerCert(commandName string, fullName string, out io.Writ
 	flags.StringVar(&options.KeyFile, "key", "", "The key file. Choose a name that indicates what the service is.")
 
 	flags.StringSliceVar(&options.Hostnames, "hostnames", options.Hostnames, "Every hostname or IP you want server certs to be valid for. Comma delimited list")
-	flags.BoolVar(&options.Overwrite, "overwrite", true, "Overwrite existing cert files if found.  If false, any existing file will be left as-is.")
+	flags.BoolVar(&options.Overwrite, "overwrite", false, "Overwrite existing cert files if found.  If false, any existing file will be left as-is.")
```

Comment:
This is probably breaking. It means things that run this for cert renewal/rotation will break

Reply:
I am trying to understand the logic behind this.
Is this affecting all kind of certs equally ?
I would definitely be ok overwriting server certs, everything in the infra ideally needs to be able to work when a cert rotates.
However if this utility also deal woth the CA cert or other root-of-trust related certs, then a default overwrite is a sharp blade wth no handle.

Reply:
So the help text from current master:

```
$ oc adm ca create-server-cert -h
Create a key and server certificate 

Create a key and server certificate valid for the specified hostnames, signed by the specified CA. These are useful for
securing infrastructure components such as the router, authentication server, etc. 

Example: Creating a secure router certificate. 

  CA=openshift.local.config/master
      oc adm ca create-server-cert --signer-cert=$CA/ca.crt \
            --signer-key=$CA/ca.key --signer-serial=$CA/ca.serial.txt \
            --hostnames='*.cloudapps.example.com' \
            --cert=cloudapps.crt --key=cloudapps.key
  cat cloudapps.crt cloudapps.key $CA/ca.crt > cloudapps.router.pem

Usage:
  oc adm ca create-server-cert [options]

Options:
      --cert='': The certificate file. Choose a name that indicates what the service is.
      --expire-days=730: Validity of the certificate in days (defaults to 2 years). WARNING: extending this above default value is highly discouraged.
      --hostnames=[]: Every hostname or IP you want server certs to be valid for. Comma delimited list
      --key='': The key file. Choose a name that indicates what the service is.
      --overwrite=true: Overwrite existing cert files if found.  If false, any existing file will be left as-is.
      --signer-cert='openshift.local.config/master/ca.crt': The certificate file.
      --signer-key='openshift.local.config/master/ca.key': The key file.
      --signer-serial='openshift.local.config/master/ca.serial.txt': The serial file that keeps track of how many certs have been signed.

Use "oc adm options" for a list of global command-line options (applies to all commands).
```

```
$ oc adm ca create-signer-cert -h 
Create a self-signed CA key/cert for signing certificates used by server components.

Usage:
  oc adm ca create-signer-cert [options]

Options:
      --cert='openshift.local.config/master/ca.crt': The certificate file.
      --expire-days=1825: Validity of the certificate in days (defaults to 5 years). WARNING: extending this above default value is highly discouraged.
      --key='openshift.local.config/master/ca.key': The key file.
      --name='openshift-signer@1517594420': The name of the signer.
      --overwrite=true: Overwrite existing cert files if found.  If false, any existing file will be left as-is.
      --serial='openshift.local.config/master/ca.serial.txt': The serial file that keeps track of how many certs have been signed.

Use "oc adm options" for a list of global command-line options (applies to all commands).
```

I guess it may be OK to overwrite since it is not the root CA bits?

Reply:
Yeah, sounds like overwrite = true is not the end of the world here.
So perhaps it is better to keep backwards compatibility

Reply:
I agree with leaving overwrite defaulting to true. the rest looks OK to me